### DISCUSSION TITLE
Level 4 1차 데모데이까지의 방향성

### DISCUSSION THREAD

**Discussion by bigsaigon333** [woowacourse-teams/2021-gpu-is-mine#344] [General]
Level 4 1차 데모데이까지 하고 싶은 일, 어떻게 진행하고 싶은지 등 자유롭게 적어주세요

**Comment by MyaGya** [comment]
- README 작성
- 공지사항 페이지
- 공지사항을 적을 수 있도록 관리자 페이지 작성
- GM 정해진 URL대로 도작하도록 변경
- 서비스, 패키지 구조 리펙터링

**Reply by bigsaigon333** [reply]
일반 사용자에게 오픈할 경우에 사용법, 서비스 소개 등을 자세히 적어서 알릴 필요가 있다. 

완태: README.md 에 대한 링크를 달면 어떨까?
동동: notion 에다가 깔끔하게 쓰는 것도 좋을듯

**Comment by sjpark-dev** [comment]
- 다른 사람이 구현한 기능을 이해해서 내 것으로 만들기
- 미션을 진행하면서 생기는 인사이트를 프로젝트에 반영하기
- 서비스 구조 리팩터링
- 테스트 코드 최적화

**Comment by wannte** [comment]
1. elk 사용을 발전시킨다. (다음 두 가지 정도의 방법이 있을듯)
- 현재 방식을 유지하되, 모니터링을 도입하여 관리가 가능하게끔.
- 관리가 용이해지게, NoSql이나, File을 도입해본다.
2. 시간 부분 리팩토링을 진행한다.
3. 로그의 실시간 정보 제공을 위한 최적화 방법을 고민한다.
- 매번 모든 정보를 다 보내주는데, 이를 개선한다.(Redis Cache를 사용해본다?)

**Comment by bigsaigon333** [comment]
1. 성능 최적화
- Level 4 FE 미션으로 성능 최적화를 하고 있는데, 이 부분을 팀 프로젝트에도 적용해보고 싶습니다. 
  (코드 스플리팅, 번들 사이즈 최적화 등)
2. 테스트 코드 추가
-  테스트 할 수 있는 모든 코드에 다 테스트 코드 추가
3. 포코, 콜린과 기술 회의
**예상 안건**
-  상태 관리 라이브러리 도입 여부 검토
- ts-loader > babel-loader로 전환
4. README 작성

**Comment by 2SOOY** [comment]
+ 프로젝트 전반
   + README
   + 소개 페이지
+ UI 관련 추가하거나 바꿀 부분 생각해보기
   + gpu및 job 상세 조회의 시간 정보 추가하기
   + 토큰 만료시 자동 로그아웃 필요
+ 어플리케이션 사용해보면서 필요한 기능(UI) 추가하기
+ job 상세 조회 부분 최적화 필요

**Comment by ecsimsw** [comment]
1. README 작성
2. DAO 성능 확인 
3. NPE 위험 사항 확인 및 개선
4. 빠진 예외 명시 (IllegalArgumentException으로 던진 것들?)

**Comment by knae11** [comment]
- README 작성
- ADMIN 페이지 main 서버에도 올리기
- 프로젝트 사용자 방향성을 어떻게 가져갈 것인지 고민해보기(EC2에 올리고 토이플젝처럼 활용한다는게 어떤 의미인지 아직 잘 모르겠음)
    - lab 에 대한 권한이 필요없어질 듯?! (더불어 manager, admin 의 권한도 필요없어지지 않을까? 이런 권한 문제를 좀 명확히 하고 싶음) 
- 프로젝트 로그 관련해서 설정해둔 파일 알아보면서 모니터링 방안을 좀 더 개선할 방향 없는지 확인해 보고 싶음
- [개인공부] 기존 프로젝트 내용 온전히 이해하기! -> 개선 방향성과 깊은 내용 더 공부해보기
    - DB partition
    - 순환참조관계
    - ELK 에서 Cluster 동작방식에 대한 이해
    - 무중단배포와 쿠버네티스
    - sonarQube, jacoco 등 설정파일
    - JPA 공부좀 해야할 것 같은데 뭘 공부해야 될 지 잘 모르겠음!ㅎㅎㅎㅎㅎㅎㅎ
- 여유가 안 될 것 같은데 혹~~~~~~~~시나 된다면 테스트코드 최적화 방식 고민
